demo/android/STT/porcupine-stt-demo-app/src/main/java/ai/picovoice/porcupine/sttdemo/MainActivity.java

```diff
@@ -0,0 +1,320 @@
+/*
+    Copyright 2021 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is
+    located in the "LICENSE" file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the
+    License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+    express or implied. See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package ai.picovoice.porcupine.sttdemo;
+
+
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.CountDownTimer;
+import android.speech.RecognitionListener;
+import android.speech.RecognizerIntent;
+import android.speech.SpeechRecognizer;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.ScrollView;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.widget.ToggleButton;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+
+import java.util.ArrayList;
+import java.util.Locale;
+
+import ai.picovoice.porcupine.Porcupine;
+import ai.picovoice.porcupine.PorcupineException;
+import ai.picovoice.porcupine.PorcupineManager;
+import ai.picovoice.porcupine.PorcupineManagerCallback;
+
+
+public class MainActivity extends AppCompatActivity {
+    private PorcupineManager porcupineManager = null;
+
+    private ScrollView intentWrapper;
+    private TextView intentTextView;
+    private TextView stateTextView;
+
+    private SpeechRecognizer speechRecognizer;
+    private Intent speechRecognizerIntent;
+
+    private void displayError(String message) {
+        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
+    }
+
+    private PorcupineManager initPorcupine() throws PorcupineException {
+        final Spinner mySpinner = findViewById(R.id.keyword_spinner);
+        final String keywordName = mySpinner.getSelectedItem().toString();
+
+        stateTextView.setText("Listening for Wake Word");
```

Comment:
Since you have keyword name here, maybe the text should be "Listening for '{keyword name}'"